workspaces/ui-v2/src/components/DebugOpticComponent.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC, useEffect } from 'react';
+
+import { IOpticSpecRepository } from '@useoptic/spectacle/build';
+
+declare global {
+  interface Window {
+    __GLOBAL_DIFF_DEBUG_FUNCTION__?: () => any;
+    debugOptic?: () => void;
+  }
+}
+
+const GLOBAL_DIFF_DEBUG_FUNCTION = '__GLOBAL_DIFF_DEBUG_FUNCTION__';
+
+// In the future, we could directly dump the redux store
+export const useGlobalDiffDebug = (fn: () => any) => {
+  useEffect(() => {
+    window[GLOBAL_DIFF_DEBUG_FUNCTION] = fn;
+    return () => {
+      delete window[GLOBAL_DIFF_DEBUG_FUNCTION];
+    };
+  }, [fn]);
+};
+
+const debugDump = (specService: IOpticSpecRepository) => {
+  return async function () {
+    const events = await specService.listEvents();
+    const diffDebugFunction = window[GLOBAL_DIFF_DEBUG_FUNCTION];
+    const diffState = diffDebugFunction ? diffDebugFunction() : false;
```

Comment:
Thoughts on just dumping the diffState? I think it's probably the most useful since those are inputs to the UI - such that if the UI crashes we know why it crashed + what interaction. Is this enough information for us to debug the diff flow (i.e. everything in https://github.com/opticdev/optic/blob/develop/workspaces/ui-v2/src/pages/diffs/contexts/SharedDiffContext.tsx)?

Reply:
I think it's a good starting point

Reply:
I don't think it would give us the interaction